cmd/federation-controller/main.go

```diff
@@ -210,56 +202,34 @@ func main() {
 		if err != nil {
 			log.Fatalf("failed to create FDS client: %v", err)
 		}
-		startFDSClient = func() {
-			go func() {
-				if err := federationClient.Run(); err != nil {
-					log.Errorf("failed to start FDS client, will reconnect in %s: %v", retryDelay, err)
-					time.AfterFunc(retryDelay, federationClient.Restart)
-				}
-			}()
-		}
 	}
 
-	if cfg.ConfigMode == config.ConfigModeMCP {
-		mcpServer := adss.NewServer(
-			&adss.ServerOpts{Port: 15010, ServerID: "mcp"},
-			meshConfigPushRequests,
-			startFDSClient,
-			mcp.NewGatewayResourceGenerator(istioConfigFactory),
-			mcp.NewServiceEntryGenerator(istioConfigFactory),
-			mcp.NewWorkloadEntryGenerator(istioConfigFactory),
-			mcp.NewVirtualServiceResourceGenerator(istioConfigFactory),
-			mcp.NewDestinationRuleResourceGenerator(istioConfigFactory),
-		)
-		if err := mcpServer.Run(ctx); err != nil {
-			log.Fatalf("Error running XDS server: %v", err)
-		}
-	} else if cfg.ConfigMode == config.ConfigModeK8s {
-		istioClient, err := istiokube.NewClient(istiokube.NewClientConfigForRestConfig(kubeConfig), "")
-		if err != nil {
-			log.Fatalf("failed to create Istio client: %v", err)
-		}
-
-		rm := kube.NewReconcilerManager(
-			meshConfigPushRequests,
-			kube.NewGatewayResourceReconciler(istioClient, istioConfigFactory),
-			kube.NewServiceEntryReconciler(istioClient, istioConfigFactory),
-			kube.NewWorkloadEntryReconciler(istioClient, istioConfigFactory),
-			kube.NewVirtualServiceReconciler(istioClient, istioConfigFactory),
-			kube.NewDestinationRuleReconciler(istioClient, istioConfigFactory),
-		)
-		if err := rm.ReconcileAll(ctx); err != nil {
-			log.Fatalf("initial Istio resource reconciliation failed: %v", err)
-		}
-		go rm.Start(ctx)
+	istioClient, err := istiokube.NewClient(istiokube.NewClientConfigForRestConfig(kubeConfig), "")
+	if err != nil {
+		log.Fatalf("failed to create Istio client: %v", err)
+	}
 
-		if startFDSClient != nil {
-			log.Info("Starting FDS client")
-			startFDSClient()
-		}
+	rm := kube.NewReconcilerManager(
+		meshConfigPushRequests,
```

Comment:
It doesn't need to be addressed here, but currently only the `TypeUrl` of the push request is used in k8s mode. The type of the `meshConfigPushRequests` channel could now just be `chan string`.

Reply:
Thanks for pointing this out. I prefer to address this comment in a follow-up PR, because it will require little more refactoring in components, which takes this channel as an argument.